src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -210,6 +210,56 @@ String decrypt(final String encryptedCredential) {
         return new String(decryptedResult.getResult(), StandardCharsets.UTF_8);
     }
 
+    /**
+     * Removes the credential fields from a template
+     * @param template the template
+     * @return the redacted template
+     */
+    public Template redactTemplateCredentials(Template template) {
+        Template.Builder processedTemplateBuilder = new Template.Builder();
```

Comment:
(Style) Since the method is `redactFoo()` it might make sense if the internal variables were `redactedBar` rather than `processedBar`.